src/lib/business/UserProfileManager.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * User Profile Manager
+ * This class is responsible for managing the user profile data
+ */
+export class UserProfileManager {
+  private supabase: any
+  private session: any
+
+  constructor(supabase: any, session: any) {
+    this.supabase = supabase
```

Comment:
We should keep strong typing for these if we want to take this.

Reply:
Totally agree, I just didn't find the class to import.

Reply:
See https://github.com/CriticalMoments/CMSaasStarter/pull/108#issuecomment-2254491650 before working on it though, should come up with the strategy!